sourcegraph/sourcegraph
# vscode: add experimental esbuild support

Closes #28823
 
Adds ability to build the VS Code extension with esbuild. Production build time has gone from 100 seconds to 1.5 seconds on my M1 Pro MBP! Moves and modifies experimental web esbuild plugins to work with the VS Code extension.

| esbuild | webpack |
| --- | --- |
| <img width="344" alt="Screen Shot 2022-04-18 at 10 55 36 AM" src="https://user-images.githubusercontent.com/37420160/163827354-0065ce77-1c99-40a0-8e64-f507a909433d.png"> | <img width="359" alt="Screen Shot 2022-04-18 at 12 03 48 PM" src="https://user-images.githubusercontent.com/37420160/163836682-6c185d90-b99b-422f-86ba-2c4f71a6ab08.png"> |

![Screen Shot 2022-04-18 at 10 55 22 AM](https://user-images.githubusercontent.com/37420160/163827375-2423b5ae-0d7d-476a-9c9a-16da8fb77913.png)



TODO: 
- ensure no regressions with integration tests (create separate issue)


## Test plan

- Steps for node extension host
  - Run `yarn build:esbuild` in the `client/vscode` directory
  - Run `"Launch VS Code Extension"` task in VS Code
  - Note that search and remote file browsing still work
- Steps for webworker extension host
  - Run `yarn build:esbuild:web` in the `client/vscode` directory
  - Run `"Launch VS Code Web Extension"` task in VS Code
  - Note that search and remote file browsing still work

## App preview:

- [Link](https://sg-web-tjk-vsce-esbuild.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-evixyevufh.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
